tests/test_file_watch.rb

```diff
@@ -55,6 +55,25 @@ def test_events
       assert($deleted)
       assert($unbind)
     end
+
+    # Refer: https://github.com/eventmachine/eventmachine/issues/512
+    def test_invalid_signature
+      # This works fine with kqueue, only fails with linux inotify.
+      omit_if(EM.kqueue?)
+
+      EM.run {
+        file = Tempfile.new('foo')
+
+        w1 = EventMachine.watch_file(file.path)
+        w2 = EventMachine.watch_file(file.path)
```

Comment:
This returns a real Watcher object that isn't valid? That seems bad, like, the better solution here is to return `nil` if the file is already being watched and we can't watch it a second time.

In the case of kqueue, does it actually create two independent watchers?